lib/lbt/resources/ResourcePool.js

```diff
@@ -112,14 +112,15 @@ class ResourcePool {
 		this.whenReady = Promise.resolve(true);
 	}
 
-	addRoot(fileOrFolder, prefix) {
+	async addRoot(fileOrFolder, prefix) {
 		this._roots.push({
 			root: fileOrFolder,
 			prefix: prefix
 		});
 		log.verbose("  scanning <%s> for resources (prefix='%s')", fileOrFolder, prefix);
 		const p = scanFileOrDir(fileOrFolder, prefix, this);
 		this.whenReady = this.whenReady.then( () => p );
+		return this.whenReady;
```

Comment:
But then you never know when the function succeeded, because `scanFileOrDir` is async otherwise I wasn't able to test it and scanFileOrDir is private (cannot be mocked)

Reply:
exactly, otherwise I wasn't able to determine when the method finished